portfolio/src/main/java/com/google/sps/data/Task.java

```diff
@@ -13,32 +14,59 @@
   private final Optional<String> description;
   private final Duration duration;
   private final TaskPriority priority;
+  private Instant scheduledTime;
 
   // TODO(raulcruise): Add an ID field if necessary.
 
   /**
    * The constructor will make sure that all necessary parameters are passed in, and populate each
    * class variable appropriately. None of the fields can be null with exception to the description
    * as it is optional. If the description is passed in as null then it will be stored as an empty
-   * Optional object.
+   * Optional object. Tasks by default do not have a scheduled time, they are only assigned this
+   * field as a result of getting scheduled by the scheduling algorithm.
    */
-  public Task(String name, String description, Duration duration, TaskPriority priority) {
+  public Task(String name, String description, long durationMinutes, TaskPriority priority) {
     if (name == null) {
       throw new IllegalArgumentException("Name cannot be null");
     }
 
-    if (duration == null) {
-      throw new IllegalArgumentException("Duration cannot be null");
+    if (priority == null) {
+      throw new IllegalArgumentException("Priority cannot be null");
+    }
+
+    this.name = name;
+    this.description = Optional.ofNullable(description);
+    this.duration = Duration.ofMinutes(durationMinutes);
+    this.priority = priority;
+  }
+
+  /**
+   * The constructor allows for Tasks to have their scheduled time be set. New tasks being made for
+   * scheduling purposes use this constructor instead.
+   */
+  public Task(
```

Comment:
Can we use param comments? For example, they can indicate which fields can be null and which cannot be.

Reply:
Resolved.